# Task no.1
***

### Newton's method : Python script to estimate the square root of a float number inserted by the user

Read input number and estimate relevant info (no. decimal places)

In [1]:
no_decimal=0

number= float(input("Insert the desired float number you wish to square root ")) # ask the user the float number
no_decimal= len(str(number).split(".")[1])  # count the number of decimal places of the inserted float number



print ("No of decimal places {} \n\n\n\n".format(no_decimal))


ValueError: could not convert string to float: ''

Define the function  that estimate the square root throught Newton's method

In [ ]:
def my_sqrt(number,tolerance):
    guess = number/2 # first trial for the guessing. Iterate further for approximate the value 

    value= (guess + number/guess)/2 # estimate a new value
    
    while abs(value - guess)>tolerance: 
        guess = value # guess upadated with the new estimate value
        value = (guess + number/guess)/2
    return guess




Run the previous created function and print its original estimation

In [ ]:
tolerance=0.0001
root=my_sqrt(number,tolerance)
print("The estimated square root for the inserted number:{} is {}\n\n\n\n".format(number, root))


The estimated square root for the inserted number:1.2 is 1.0954451212048808






Trim the original estimation for the squared root value and make it agredable with the no. of inserted decimal places of the input number

In [ ]:

root_decimal= x = round(root, no_decimal)

print("The estimated square root with {} decimal places is {}\n\n\n\n".format(no_decimal,root_decimal))


The estimated square root with 1 decimal places is 1.1






# Task no.2
***

### Contingency table 


With the user drink and food preferences a data frame  is constructed. 
<ul>
 <li>Number of people liking Coffee and chocolate biscuits = 43</li>
 <li>Number of people liking Coffee and plain biscuits = 57</li>
  <li>Number of people liking Tea and chocolate biscuits = 56</li>
   <li>Number of people liking Tea and plain biscuits = 45</li>
  </ul>


In [2]:
import pandas as pd

coffee_choco=[['Coffee','Chocolate']]*43
coffee_plain=[['Coffee','Plain']]*57
tea_choco=[['Tea','Chocolate']]*56
tea_plain=[['Tea','Plain']]*45

preferences=coffee_choco+coffee_plain+tea_choco+tea_plain

drink, food = list(zip(*preferences))

df = pd.DataFrame({'drink': drink, 'food': food})



Create a contingency table with the created dataframe so it would be readable by scipy.stats (python library)

In [7]:
import scipy.stats as ss

cross = ss.contingency.crosstab(df['drink'], df['food'])
cross.count

array([[43, 57],
       [56, 45]])

Perform a chi-square to the data throught the means of a contingency/frequency table and evaluate the association between both preferences: food and drink.
Chi-square is a statistical hypothesis test that consideres a significance level (p-value) for the rejection/acceptance of null/alternative hypothesis. 
<ul>
 <li>The null hypothesis, commonly refered as &mu;<sub>0</sub> ditactes that there is no association between the tested categorical variables. </li>
 <li>The alternative hypothesis, commonly refered as &mu;<sub>1</sub> ditactes that there is association between the tested categorical variables. </li>
</ul>
The considered p-value was 0.05, thus if p-value is <<<< than 0.05 means that &mu;<sub>0</sub> must be rejected and there is evidence of association between variables. In other hand, if p-value is greater thatn 0.05 the null hyphotesis must be taken meaning no association between both variables: food and drink. 


In [8]:
chi2_stat, p_val, dof, expected = ss.chi2_contingency(cross.count)
p_val

0.10447218120907394

p-value= 0.104 which means that null hyphotesis can not be rejected, so there is no evidence of association between food prefernce and drink preference

# Task no. 3

In [9]:
import pandas as pd
from scipy import stats



# Upload data set
data = pd.read_csv('penguins.csv')
#data.head() # to display the first 5 lines of loaded data



# Prepare the data set - remove the nan
data_clean = data.dropna()

# Isolate the data-set per sex
male_penguins = data[data['sex'] == 'MALE']
female_penguins = data[data['sex'] == 'FEMALE']

# Restrict the analysis exclusively to the bodymass
body_male=male_penguins["body_mass_g"]
body_female=female_penguins["body_mass_g"]
#print(body_male)
#print(body_female)


t_stat, p_value=stats.ttest_ind(body_male, body_female)


#Null hypothesis: H0: μ_male = μ_female


# Determine significance
alpha = 0.05
if p_value < alpha:
    print("There is significant evidence to reject the null hypothesis. This means that male a female groups have different body mass averages")
else:
    print("There is no significant evidence to reject the null hypothesis.This means that female and male can have different or equal body mass averages")



There is significant evidence to reject the null hypothesis. This means that male a female groups have different body mass averages


# Task no. 4

In [10]:
import pandas as pd
from scipy import stats



# Upload data set
data = pd.read_csv('iris.csv')
data.head() # to display the first 5 lines of loaded data



sepal_length  sepal_width  petal_length  petal_width   class
0           5.1          3.5           1.4          0.2  setosa
1           4.9          3.0           1.4          0.2  setosa
2           4.7          3.2           1.3          0.2  setosa
3           4.6          3.1           1.5          0.2  setosa
4           5.0          3.6           1.4          0.2  setosa